generator.rb

```diff
@@ -0,0 +1,66 @@
+# Make sure user inputted argument is long enough. If it's not, reprompt the user. Return new (or original, if already okay) input.
+def check_length(word)
+  until word.length > 1
+    print "Your word must be at least 2 characters long. > "
+    word = gets.chomp
+  end
+  return word
+end
+
+# Check if a letter is a vowel. If it is, return true. If not, return false.
+def is_vowel?(letter)
+  case letter
+  when "a", "e", "i", "o", "u"
+    return true
+  else
+    return false
+  end
+end
+
+# Returns an integer of how many characters there are after the first vowel in an argument.
+def segment_length(word)
+  chars_before_vowel = 0
+  word.each_char do |letter|
+    if is_vowel?(letter)
+      break
+    end
+    chars_before_vowel += 1
+  end
+  length = word.length - chars_before_vowel
+  return length
+end
+
+# When given a word and a word number (1 or 2, depending on whether it's the first or second word in the Portmanteau), returns the correct word segment.
+def make_portmanteau(word, word_number)
```

Comment:
This method is FASCINATING and WONDERFUL! Great work finding a way to make this complex logic into a method that works for either the first or second word.